File: examples/autovalue-example/src/main/java/nz/bradcampbell/autovalueexample/DateTypeAdapter.java
package nz.bradcampbell.autovalueexample;

import android.os.Parcel;
import java.util.Date;
import nz.bradcampbell.paperparcel.TypeAdapter;
import org.jetbrains.annotations.NotNull;

public class DateTypeAdapter implements TypeAdapter<Date> {
  @NotNull @Override public Date readFromParcel(@NotNull Parcel inParcel) {
    return new Date(inParcel.readLong());
  }

  @Override public void writeToParcel(@NotNull Date value, @NotNull Parcel outParcel, int flags) {
    outParcel.writeLong(value.getTime());
  }
}

File: examples/kotlin-example/src/main/kotlin/nz/bradcampbell/paperparcel/kotlinexample/DateTypeAdapter.kt
package nz.bradcampbell.paperparcel.kotlinexample

import android.os.Parcel
import nz.bradcampbell.paperparcel.TypeAdapter
import java.util.*

object DateTypeAdapter : TypeAdapter<Date> {
  override fun writeToParcel(value: Date, outParcel: Parcel, flags: Int) {
    outParcel.writeLong(value.time)
  }

  override fun readFromParcel(inParcel: Parcel): Date {
    return Date(inParcel.readLong())
  }
}
